recipes/stlab/all/conanfile.py

```diff
@@ -11,48 +11,196 @@ class Stlab(ConanFile):
+        "coroutines": [True, False],
+        "task_system": ["portable", "libdispatch", "emscripten", "pnacl", "windows", "auto"],
     }
+
     default_options = {
-        'coroutines': False
+        "boost_optional": False,
+        "boost_variant": False,
+        "coroutines": False,
+        "task_system": "auto",
     }
 
     no_copy_source = True
     _source_subfolder = 'source_subfolder'
 
-    requires = 'boost/1.69.0'
+    def _use_boost(self):
+        return self.options.boost_optional or self.options.boost_variant
+
+    def _requires_libdispatch(self):
+        # On macOS it is not necessary to use the libdispatch conan package, because the library is 
+        # included in the OS.
+        return self.options.task_system == "libdispatch" and self.settings.os != "Macos"
+
+    def requirements(self):
+        if self._use_boost():
+            self.requires("boost/1.75.0")
+
+        if self._requires_libdispatch():
+            self.requires("libdispatch/5.3.2")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = "libraries-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
-    def configure(self):
+    def _configure_boost(self):
+        self.options["boost"].shared = True
+
+        self.options["boost"].without_atomic = True
+        self.options["boost"].without_chrono = True
+        self.options["boost"].without_container = True
+        self.options["boost"].without_context = True
+        self.options["boost"].without_contract = True
+        self.options["boost"].without_coroutine = True
+        self.options["boost"].without_date_time = True
+        self.options["boost"].without_exception = True
+        self.options["boost"].without_fiber = True
+        self.options["boost"].without_filesystem = True
+        self.options["boost"].without_graph = True
+        self.options["boost"].without_graph_parallel = True
+        self.options["boost"].without_iostreams = True
+        self.options["boost"].without_json = True
+        self.options["boost"].without_locale = True
+        self.options["boost"].without_log = True
+        self.options["boost"].without_math = True
+        self.options["boost"].without_mpi = True
+        self.options["boost"].without_nowide = True
+        self.options["boost"].without_program_options = True
+        self.options["boost"].without_python = True
+        self.options["boost"].without_random = True
+        self.options["boost"].without_regex = True
+        self.options["boost"].without_serialization = True
+        self.options["boost"].without_stacktrace = True
+        self.options["boost"].without_system = True
+        self.options["boost"].without_test = True
+        self.options["boost"].without_timer = True
+        self.options["boost"].without_thread = True
+        self.options["boost"].without_type_erasure = True
+        self.options["boost"].without_wave = True
```

Comment:
This should be left out.
Not needing any libraries or be happy with a header-only libraries can be configured in a profile.

Adding these option overrides here can clash with another profile requiring e.g. boost filesystem.

Reply:
agree, please remove, it will cause lots of troubles to consumers

Reply:
bump this needs to be removed